app/views/comments/_form.html.erb

```diff
@@ -0,0 +1,36 @@
+<%= form_with(model: @comment) do |form| %>
+    
+    <% if @comment.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@comment.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% @comment.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
```

Comment:
Move the logic to a helper. There should be absolutely no logic  in view. Views are only use to render information that became available to them from it's corresponding controller and not for generating it.